### ISSUE TITLE
Unix sockets are not implemented

### ISSUE THREAD

**Issue by guineawheek** [Microsoft/BashOnWindows#134]
socket() works, but bind() doesn't create the actual socket file.
This kills things like wine and ssh-agent.

**Comment by SvenGroot** [comment]
Unix sockets are partially implemented, and they do work. The socket files just don't show up in the file system. However, it's still possible to connect to them if you know the path of the socket file. Of course, this does break applications that attempt to "stat" a socket file or do anything else that depends on the file being present.

We're working to address this issue.

**Comment by sunilmut** [comment]
As @SvenGroot mentions, file system integration of UNIX sockets is not complete yet. We will be tracking\prioritizing which features to light up next by looking at the feedback provided at the wpdev portal. We appreciate your feedback and please make sure to provide it through the wpdev portal as well.

**Comment by MadcowD** [comment]
@SvenGroot Just an update I can reproduce the issue referenced with #1331. Jupyter notebook is using sockets and getting Invalid argument (src/tcp_address.cpp:171)

**Comment by benhillis** [comment]
Adding link to source code exhibiting an issue: https://github.com/Microsoft/BashOnWindows/issues/207

**Comment by lypanov** [comment]
This AFAICT also breaks Rails 5 out of the box without the hack of disabling of spring (it's simply too slow at times..). Any news on this?

**Comment by mindplay-dk** [comment]
This may be what's breaking the default socket-based configuration with NGINX and PHP-FPM? I was able to get it to run reliably only after configuring it to use TCP instead of sockets, and also [failed](https://github.com/Microsoft/BashOnWindows/issues/1851#issuecomment-292907880) to use sockets with PHP, is this all related to this issue?

Is there a fix on the way, or is this very far off? I really want to use UOW for development on my Windows system, mainly because there are features (such as sockets, threads and readline support) that are unavailable in Windows-builds of PHP.

**Comment by aseering** [comment]
Could someone on the WSL team perhaps summarize what they believe to be the status of this issue?

I have personally used UNIX Sockets without any issue.  But I haven't done anything fancy with them.  I realize it's more complicated than a simple "works" vs "not works" -- there are lots of flags and options and whatnot that you can set on a socket that affect its behavior; my understanding is that some work and some do not.

**Comment by sunilmut** [comment]
@aseering - We are not aware of any open issues with the WSL AF_UNIX socket implementation. But, I know it's not fully compatible yet because for example the `dbus` test does not give the same results as that of native Ubuntu. We haven't dug into all the failures yet. We are trying to prioritize issues reported here as they come.

@mindplay-dk - Thanks for your post. We will have to take one issue at a time, so that we can root cause and fix it. Can you please post repro steps, build number and strace, if possible? You can see the template [here](https://github.com/Microsoft/BashOnWindows/blob/master/ISSUE_TEMPLATE.md). To gather strace, use `strace -o <trace_file> -ff <cmd>`

**Comment by aseering** [comment]
Regarding capturing an strace, the challenge here will presumably be identifying what command to run.  (The "command" in this case is two separate backgrounded system services, nginx and php-fpm; both would need to be run directly and in the foreground.)  @mindplay-dk -- I think you indicated on another ticket that you weren't sure how to accomplish that level of Linux magic?  I can try to take a look, but I may not get to it for a while.  If anyone else is following this thread and would be willing to give it a shot, please do.

**Comment by aseering** [comment]
@sunilmut -- thanks!  The title of this ticket implies that UNIX sockets don't work at all.  And the ticket is still open.  Up to you, but I wonder if some ticket-wrangling might be in order? :-)

**Comment by sunilmut** [comment]
@aseering - Makes sense. I will close this issue out, it's too broad and not actionable. @mindplay-dk - We will continue to work with you on the other issue.

As for the strace, we would have to then rely on the repro.

**Comment by mindplay-dk** [comment]
@sunilmut I know you closed this issue, but not sure where else to answer the questions you asked. (?)

To help you repeat the php-pm issue, here's the sandbox project for testing:

https://bitbucket.org/mindplaydk/http-bench

This project has a bunch of different scripts that all do the same thing - return a JPEG image - in different ways, because I'm using it to benchmark and evaluate the performance of various means of sending an image.

Clone the project locally.

Run `composer install` and you should be all set.

Run `vendor/bin/ppm` from the project root folder to launch.

The relevant changes for the php-pm setup happened in [this commit](https://bitbucket.org/mindplaydk/http-bench/commits/fe6bfeefca30bf541d430ec82448341e6dd36b5f) and the only thing I wrote really was [this simple front controller](https://bitbucket.org/mindplaydk/http-bench/src/fe6bfeefca30bf541d430ec82448341e6dd36b5f/src/Bootstrap.php?at=master&fileviewer=file-view-default) which does nothing except return the image, it doesn't even care what the Request looks like at all. Quick and dirty.

I posted steps to recreate my environment [here](https://github.com/Microsoft/BashOnWindows/issues/1851#issuecomment-292805212).

I'm on Windows 10 Pro version 1703 build 15063.13

I tried to run that "strace" you asked for, so, this:

    strace -o ./strace -ff vendor/bin/ppm start

It produced [these files](https://github.com/Microsoft/BashOnWindows/files/914300/strace.zip).

I think that's everything you asked for - let me know if there's anything else I can do?

**Comment by mindplay-dk** [comment]
(oh, and I guess `composer install` worked on my system because it's in the path on my Windows system already and the `phar` file works with the Linux installation of PHP as well. If you don't have Composer installed, you may need to [do that first](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-osx).)

**Comment by awesomelewis2007** [comment]
im getting a  Socket type not supported when i use tcpdump anyone help
